SalesforceCommerceCloud/pwa-kit
# @W-13974739 Showing strikethrough price on PLP

# Description
This PR implemented strikethrough price for products on PLP if any product has a listing sale price 
![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/cb18c0c0-0426-4d26-8dd3-5706eedee0d4)

NOTE: this PR does not take into account changes needed for PDP



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Checkout the code
- npm ci at root
- start retail app
- go to PLP http://localhost:3000/global/en-GB/category/mens-accessories-ties or http://localhost:3000/global/en-GB/category/mens-clothing-suits to see that original price is showing with a strike through line. 
- Go to a PDP, and check the recommended section and recently view section. Make sure that the produc tiles also display strikethrough prices if applicable.

There will be some differences between showing price for variant product PLP Product Tile and Recommendation Section Product Tile due to the data differences

- For PLP, we call getProductSearch which will return data that has represent_product that we will use to display price
- For recommendation section, it is not the case because we use getProducts to retrieve data that does not have `represent_product`. In this case we will show the price of the first variant (if applicable)
- For other product types like standard/set/bundles, they will be the same for both

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
